DataOnion/Auth/IAuthServiceStrategy.cs

```diff
@@ -0,0 +1,9 @@
+namespace DataOnion.Auth
+{
+    internal interface IAuthServiceStrategy<T>
+    {
+        Task<T?> LoginAsync(T userSession);
+        Task<T?> GetSessionAsync(string id);
+        Task LogoutAsync(string id);
+    }
```

Comment:
can we just put this in the same file as its implementatiom? I put them in separate files when the interface is extensive but i think this can just be in the same file